doc/devbox_setup.md

```diff
@@ -0,0 +1,115 @@
+# Prepare your development environment
+
+This document describes how to prepare your development environment to build and use the **Microsoft Azure IoT device client SDK for .NET**
+
+1.  [Prerequisites required to build the SDK](#min_setup)
+2.  [Optional prerequisites required to test Xamarin, Windows IoT and .NET Micro Framework](#advanced)
+3.  [Test prerequisites](#testprereq)
+
+<a name="min_setup"/>
+
+## Prerequisites required to build the SDK
+
+The following is the minimum required to build and produce IoT SDK binaries. 
+This may not be sufficient to run certain samples that require Visual Studio and other SDKs or frameworks installed on your machine.
+
+### Windows + .NET Framework
+
+- Install the latest .NET Core from https://dot.net
+- Install .NET Framework 4.7 Developer Pack: https://support.microsoft.com/en-us/help/3186612/the-net-framework-4-7-developer-pack-and-language-packs
+- Install .NET Framework 4.5.1 Developer Pack: https://www.microsoft.com/en-us/download/details.aspx?id=40772
```

Comment:
I need both 4.7 and 4.5.1 of the same developer pack?

Reply:
Yes. .NET 4.5.1 is an older version and, since we still support it, it needs to be available in order to build our binaries.
.NET 4.7 is a good compromise for testing a compliant `netstandard2.0` .NET Framework runtime.

Note that all of this document explains what needs to be available for an SDK developer, not an app-developer. App-developers should need only a single run-time or no run-time if they choose AOT compilation (which is common for devices and required on arm devices).

Reply:
Can we add a section for application developer?  So that our get-started guide can reference this section.

Reply:
It's already there: https://github.com/abhipsaMisra/azure-iot-sdk-csharp/blob/dcc9be4ccd45955d494186bf0d671b00dc5a7365/readme.md#developing-applications-for-azure-iot